cmd/jose.c

```diff
@@ -376,6 +380,37 @@ jcmd_opt_set_jwks(const jcmd_cfg_t *cfg, void *vopt, const char *arg)
     }
 }
 
+bool
+jcmd_opt_set_pem(const jcmd_cfg_t *cfg, void *vopt, const char *arg)
+{
+    bool retval;
+    json_t **jwks = vopt;
+    EVP_PKEY *pkey = EVP_PKEY_new();
+
+    if (!pkey)
+        return false;
+
+    if (!*jwks)
+        *jwks = json_array();
+
+    if (strcmp(arg, "-") == 0) {
+        if (!PEM_read_PrivateKey(stdin, &pkey, NULL, NULL))
+            return false;
+    } else {
+        FILE_AUTO *file = fopen(arg, "r");
+        // TODO: encrypted key callback for password.
+        if (!(file && PEM_read_PrivateKey(file, &pkey, NULL, NULL)))
```

Comment:
Here, it is not distinguished the case of !file vs the case of PEM_read_PrivateKey.

Distinguish between them, so that if the error has to do with PEM_read_PrivateKey, then fclose on file is done